File: restaurant-app/src/main/java/com/example/restaurant_app/entity/Category.java
// package com.example.restaurant_app.entity;
// import lombok.Data;
// import jakarta.persistence.CascadeType;
// import jakarta.persistence.Entity;
// import jakarta.persistence.FetchType;
// import jakarta.persistence.GeneratedValue;
// import jakarta.persistence.GenerationType;
// import jakarta.persistence.Id;
// import jakarta.persistence.OneToMany;

// import java.util.List;

// import org.hibernate.cache.spi.support.AbstractReadWriteAccess.Item;

// @Entity
// @Data
// public class Category {
//     @Id
//     @GeneratedValue(strategy = GenerationType.IDENTITY)
//     private Long id;

//     private String name;

//     @OneToMany(mappedBy = "category", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
//     private List<Item> items;
// }

File: frontend/src/components/Category.js
import React from 'react';
import MenuItem from './MenuItem';

const Category = ({ category }) => {
  return (
    <section>
      <h2>{category.name}</h2>
      <ul>
        {category.items.map(item => (
          <MenuItem key={item.id} item={item} />
        ))}
      </ul>
    </section>
  );
};

export default Category;
